.github/workflows/recover_s3_repository_periodic_test_production.yml

```diff
@@ -0,0 +1,135 @@
+            echo "restored folder ${{ env.TEST_FOLDER_ABS_PATH }}.restored should not exist"
+            exit 1
+          fi
+
+      - name: Get current datetime and sleep for a couple of minutes
+        run: |
+          now=$( date +"%m-%d-%Y %H:%M:%S %z" )
+          echo "INIT_DATETIME=$now" >> $GITHUB_ENV
+          sleep 120
+
+      - name: create a file in the bucket to be rolled back (this file should be deleted by the procedure)
+        run: |
+          echo "this is a test" > ${{ env.TESTING_FILE }}
+          aws s3 cp ${{ env.TESTING_FILE }} ${{ env.TEST_FOLDER_ABS_PATH }}/${{ env.TESTING_FILE }}
+          # ensure the file is there
+          TZ="UTC" aws s3 ls ${{ env.TEST_FOLDER_ABS_PATH }}/${{ env.TESTING_FILE }}
+
+      - name: Run S3 PIT restore in Production S3 for the test folder
+        run: |
+          BUCKET="${{ env.BUCKET_NAME }}" \
+          PREFIX="${{ env.MANDATORY_PREFIX }}/${{ env.TEST_FOLDER }}" \
+          TIME="${{ env.INIT_DATETIME }}" \
+          IMAGE="${{ env.IMAGE }}" \
+          AWS_PROFILE="${{ env.TEMP_AWS_PROFILE }}" \
+          make restore
+
+      - name: Ensure the perpetual file exists
+        run: |
+          TZ="UTC" aws s3 ls ${{ env.TEST_FOLDER_ABS_PATH }}/PERPETUAL_FILE.md
+
+      - name: Ensure the rollbacked file does not exist
+        run: |
+          set +e
+          TZ="UTC" aws s3 ls ${{ env.TEST_FOLDER_ABS_PATH }}/${{ env.TESTING_FILE }}
+          if [ $? -eq 0 ]; then
+              echo "The file ${{ env.TEST_FOLDER_ABS_PATH }}/${{ env.TESTING_FILE }} should have been deleted"
+              exit 1
+          fi
+
+      - name: Ensure the original with the original file exists
+        run: |
+          TZ="UTC" aws s3 ls ${{ env.TEST_FOLDER_ABS_PATH }}.original/${{ env.TESTING_FILE }}
+
+      - name: Delete .original
+        run: |
+          aws s3 rm "${{ env.TEST_FOLDER_ABS_PATH }}.original/${{ env.TESTING_FILE }}"
+          aws s3 rm "${{ env.TEST_FOLDER_ABS_PATH }}.original/PERPETUAL_FILE.md"
+          aws s3 rm "${{ env.TEST_FOLDER_ABS_PATH }}.original"
+
+      - name: Send Slack notification to OHAI
+        if: ${{ failure() }}
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "text": ":rotating_light: Dry-Run Recover S3 Repository failed :warning: :warning: :warning: @hero check <${{ env.GITHUB_JOB_URL }}> :rotating_light:"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.OHAI_SLACK_WEBHOOK }}
+          GITHUB_JOB_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+      - name: Send Slack notification to AC
+        if: ${{ failure() }}
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "text": ":rotating_light: Dry-Run Recover S3 Repository failed :warning: :warning: :warning: @hero check <${{ env.GITHUB_JOB_URL }}> :rotating_light:"
```

Comment:
Nit: I'm not sure if we should say _Dry-Run_, even if it runs in a testing folder it is actually running (and failing if we get this message! 😅 )

Reply:
Sounds reasonable :)  I'll rename it